### ISSUE TITLE
Feed may be featuring stories that are not marked as "featured"

### ISSUE THREAD

**Issue by jeremyf** [forem/forem#15613]
**Describe the bug**

The `Articles::Feeds::FindFeaturedStory` (see below) implies that we should select articles that are `featured == true`.  However, the implementation does not force a featured image.

https://github.com/forem/forem/blob/7010c8733002ff12d08051dfeff0b6270e1cd315/app/services/articles/feeds/find_featured_story.rb#L1-L27

**To Reproduce**

I'm uncertain, except to say that you may well see articles in the "featured" slot that are not in fact featured.  They just happen to have a main image.

**Expected behavior**

If the content administrators set an article as featured, I think that they would assume that the feed's featured article was `featured == true`.

The [Articles::Feeds::WeightedQueryStrategy](https://github.com/forem/forem/blob/7010c8733002ff12d08051dfeff0b6270e1cd315/app/services/articles/feeds/weighted_query_strategy.rb) selects an article that is `featured == true`, which I believe is conformant to the expectations.

**Screenshots**

Not applicable.

**Desktop (please complete the following information):**

Not applicable.

**Smartphone (please complete the following information):**

Not applicable.

**Additional context**

Given [Remove the requirement of the featured post to have an image · Issue #15292 · forem/forem](https://github.com/forem/forem/issues/15292), I believe that the "featured" story should have a `featured == true` attribute.

**Comment by amywtlin** [comment]
This does seem like a bug. Your definition of `featured` in expected behavior is exactly what I'd expect:
>If the content administrators set an article as featured, I think that they would assume that the feed's featured article was `featured == true`.

@michael-tharrington is there any historical reasons that this is the way it is right now? Or do you think this is also a bug?

**Comment by jeremyf** [comment]
I did some sleuthing and for the "feed" (or at least some aspects of the feed), the `featured == true` has not been part of the consideration.  (e.g. the now "Relevant" feed in particular).

**Comment by michael-tharrington** [comment]
@amywtlin — to be honest, I was always a bit unclear about exactly what that button did.

I honestly think it was just used to highlight an article in green on the https://dev.to/admin/content_manager/articles interface. Ben would click the button as a signifier to anyone working the "Content shift" (posting to buffer) that an article was good to share. We used to have a more comprehensive system set up for sending articles from https://dev.to/admin/content_manager/articles to Buffer — this was before the days of the forem2buffer app.

I slightly feel like clicking featured would also auto bump the featured number of a post by some amount, but I could be wrong about that.

**Comment by jeremyf** [comment]
@michael-tharrington I looked at the code and checking `featured` does nothing to the `featured_number`.

So, the question is a matter of least surprise:

```gherkin
Given I am a content moderator
When I say an article is "featured"
Then I would expect that article would/could be shown in the featured slot
```

Right now, the criteria for the featured slot in the "Relevant" feed is what is "popular" and has a main image.  I think that runs a bit counter intuitive to having a "featured" attribute for Article.

And if we don't add the `featured == true` filter, then if we implement the feature "don't require image for featured in the feed" will be whatever's popular/at the top of the list.  When then begins to ask the question "Why are we having a featured section?"

**Comment by michael-tharrington** [comment]
Oh okay I didn't actually know what the featured slot meant at all prior to this – it sounds like it's hidden in the code and so requires knowledge of the code to understand.

> "Why are we having a featured section?"

I think this featured box should just be removed because it's confusing.

Perhaps one day, we could have curated instead and add the item to an hand-picked/curated feed.

**Comment by jeremyf** [comment]
We have the concept of a pinned article.  Personally, I think we should say that if moderators have marked something as featured AND we have a featured slot, then it makes sense to put those featured things in that slot.

That is a far easier change than ripping out the featured slot (which I'm all for but which has a larger impact on the existing system and our users).

**Comment by michael-tharrington** [comment]
Well really, this button is admin exclusive right now, so it's not available to mods. But maybe we could make it do the same thing if an admin decides to click it?

Can you please explain what happens when someone features an article again? I'm struggling to understand this. It sounds like it increases the chance of it appearing in the top spot of a feed, right? But it doesn't increase the feature score (that's unconnected)? And it doesn't sure-fire put the article in the top (pinned) spot just makes it more likely to be there? 

It's tough for me to wrap my head around what it does and I'm not sure how I'd explain it to other Forem admin.

**Comment by jeremyf** [comment]
> Can you please explain what happens when someone features an article again?

What I've found is:

- In the [API](https://github.com/forem/forem/blob/4efcf8c3d3af123856ed355a963180562537f701/app/services/article_api_index_service.rb#L140-L146), if you don't search on anything, we only surface featured articles
- In our [digest emails](https://github.com/forem/forem/blob/4efcf8c3d3af123856ed355a963180562537f701/app/services/email_digest_article_collector.rb#L32-L40), if the user doesn't follow anything, we surface only featured articles

This also highlights that our API is using different logic than our non-logged in users which uses different logic than our logged in users which uses different logic than our RSS feed.

Ideally, I'd like to have a singular "I'm logged in here's the articles that are part of the feed" and likewise, if I'm not logged in I'd have consistent results.

(This inter-relates with DEV.to's churn of data so it's not obvious from the outside that this churn inconsistency exists, because there are tens of thousands of updates each day that refine the score and thus the order of posts)

**Comment by amywtlin** [comment]
@michael-tharrington: `featured` tick under `/admin/content_manager/articles` isn't being used by community success. Correct?

After a discussion with @jeremyf, we think it makes sense to remove this both in `/admin/content_manager/articles` and the backend algorithm, and rethink what Creators/admins want with "featured" in the Forems. The idea to feature posts still make sense to me, but the currently implementation pulls in too many unnecessary things. 

However, as these `featured` posts (the latest one is back in June, from Jeremy's query results) are actually used as fallback posts in 1) digest emails when users don't follow tags 2) API response when tags not specified, we do need to find a set of fallback posts for these cases, and potentially more similar scenarios that we haven't discovered.

**Comment by michael-tharrington** [comment]
@amywtlin that's right! The `featured` tick under `/admin/content_manager/articles` isn't being used by community success. I don't think it's being used by our editorial team either as they are now using [forem2buffer](https://github.com/forem/forem2buffer).

> After a discussion with @jeremyf, we think it makes sense to remove this both in /admin/content_manager/articles and the backend algorithm, and rethink what Creators/admins want with "featured" in the Forems. The idea to feature posts still make sense to me, but the currently implementation pulls in too many unnecessary things.

Awesome! I'd agree with this. Perhaps we can even find a way to lean on moderators to help us select "featured" posts. And maybe when a post is picked in this way, it can be marked as "staff" or "moderator" curated to specifically let onlookers know how the post was picked. The perks of this would be that it would likely feel good as an author to know that your post was handpicked + it would be very clear to our audience what is happening (i.e. no magic).

> However, as these featured posts (the latest one is back in June, from Jeremy's query results) are actually used as fallback posts in 1) digest emails when users don't follow tags 2) API response when tags not specified, we do need to find a set of fallback posts for these cases, and potentially more similar scenarios that we haven't discovered.

Gotcha! Yeah, that makes sense that we'd need to fill that hole.

**Comment by benhalpern** [comment]
I am going to work to close the loop on the clear purpose of "featuring" a post:

- It adds outsized weight to the a post's likelihood of appearing in peoples' Relevant feed, to work in conjunction with natural relevancy factors such as tags, followed users etc.
- Its precise impact may vary based on feed strategy and a/b testing, but admins can use it to promote content that is _generally_ useful to the community.
- It is the main action admins can use to impact a post in a positive way. It should be substantially more impactful than a thumbs up.
- It is not specifically related to "featured story" aka the top one with an image — we will seek to clarify verbiage here.

We will move forward with always having a button that admins can press to mark a post as important and valuable for the community. Further renaming and exact details around this are TBD, but we will work to make all of the above clear within the product and the docs. How individual admins choose to use the button will vary, but the impact will be consistent.

**Comment by mikeydorje** [comment]
Closing as stale/superseded by later feed changes. If this is still a problem today, please open a new issue with current repro steps and expected behavior.
